### ISSUE TITLE
Remove Connector Variants (-strict-encrypt, -secure, -normalized)

### ISSUE THREAD

**Issue by bnchrch** [airbytehq/airbyte#28996]
## Problem
Many of our connectors have hidden sibling connectors we build separately for use in Cloud.

These are 
* source-file/-secure
* source-postgres/-strict-encrypt
* source-alloydb/-strict-encrypt
* source-clickhouse/-strict-encrypt
* source-cockroachdb/-strict-encrypt
* source-db2/-strict-encrypt
* source-mongodb/-strict-encrypt
* source-mysql/-strict-encrypt
* source-mssql/-strict-encrypt
* source-oracle/-strict-encrypt
* destination-elasticsearch/-strict-encrypt
* destination-postgres/-strict-encrypt
* destination-clickhouse/-strict-encrypt
* destination-mongodb/-strict-encrypt
* destination-mysql/-strict-encrypt
* destination-mssql/-strict-encrypt
* destination-oracle/-strict-encrypt
* 

**This is a lot of unnecessary overhead**

This means when developing for each we need to
1. Make changes to the default e.g. source-postgres
2. Make changes to the variant e.g. source-postgres-strict-encrpyt
3. Test changes for the default
4. Test changes for the variant
3. Publish changes for the default
4. Publish changes for the variant

This is additional time and complexity for developers and for the pipeline that handles the automatic publishing of these

## Proposed Solution
[Document outlining the nuance](https://docs.google.com/document/d/1Zu7ERH3BnJRvQJIaissXGF07BTUBcZXGCQJZLknqIog/edit)

_We should remove these variants and use a platform environment variable `DEPLOYMENT_MODE` instead_
**Document DEPLOYMENT_MODE env variable as a contract**
Currently the environment variable is set, but it’s not clearly stated this is part of the contract of running connectors on the Airbyte platform. It should be made obvious this is an important contract that’s bearing load.

**Connectors publish pipeline**
On publish, the spec command is already run twice - once for cloud and once for OSS, each with a different value of DEPLOYMENT_MODE. The resulting specs are uploaded to newly-separate GCP buckets.

The missing piece here is on the registry generation side, which we'll need to update to use the cloud variant if it exists when generating the cloud registry.

As a result, the generated cloud_registry.json consumed by the platform contains the proper version of the spec.

**CATs can specify tests per environment**
(can happen in parallel with the publish pipeline changes)
The CAT framework is extended so different test cases can be specified for different environments (to validate check and spec commands behave correctly in different environments)

**Migrate spec mutators and -strict-encrypt connectors over to environment-sensitive connectors**
(This can happen over a longer period of time as a background task)
Sunset existing workarounds with -strict-encrypt connectors and spec mutators. Once all spec mutators are removed, remove the infrastructure code for it in the platform as well.

**Comment by evantahler** [comment]
Maybe we should close this in favor of https://github.com/airbytehq/airbyte/issues/30623?

**Comment by alafanechere** [comment]
Closing in favor of https://github.com/airbytehq/airbyte/issues/30623
